### ISSUE TITLE
Feature request: vectorised classify method

### ISSUE THREAD

**Issue by evertrol** [GOTO-OBS/goto-vegas#4]
The classify method runs for one object at a time, but does so in a loop.

Perhaps for efficiency reasons, could it accept any number of objects, i.e., `predictors` can be a matrix?
Perhaps as a separate method (`vclassify`).

This may complicate the calculation of scoring: it will have to look for a NotImplementedError and try the alternative instead.
It may also sow confusion for users to which of the methods to implement, or even both.

So I'm not sure how desirable this is, but working with numpy/Table/DataFrame, a vectorised version feels more straightforward.

**Comment by evertrol** [comment]
This is (partly) also related to commit 567501b and the Travis time-out for the tests.

The sklearn random forest classifier I'm using is heavily vectorised, and is behaving very poorly in a simply Python for-loop for individual predictors.  
Without having dived into the sklearn source code, I'm guessing there are a bunch of checks and conversions going on each time its `predict()` method is called. When the input is a Pandas dataframe or numpy recarray, that happens only once, and the total time is very short.  
In a for-loop, this happens every time, slowing it down tremendously. There *may* something else going on as well (I'm using a Pandas dataframe in the vectorised version, while I'm doing some nasty conversion from an Astropy table Row to a numpy array so that `predict()` can accept it; that may also be a cause of slow-down; see also issue #5).

Some timings from a local run:
Training set: 72K rows
Test set: 55K rows
Test with vectorised version: 0.1 seconds (!)
Test with for-loop & Row -> ndarray conversion: ~0.1 seconds *per* iteration

**Comment by andycasey** [comment]
I agree that a vectorised method is essential for practicality, but I am cautious not to confuse people about which functions need to be written (e.g., `vclassify` or `classify`). 

I've updated the scoring function to treat the `classify` function to accept vectors by default. I've also updated the README in the master branch, and the `classify` function on baseline.

I'm closing this and #10 for now, and if need be I will update rol-randomforest to run with the standard scoring functionality on master.
